src/main/java/seedu/address/model/person/lab/LabList.java

```diff
@@ -0,0 +1,134 @@
+package seedu.address.model.person.lab;
+
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.util.Iterator;
+import java.util.List;
+
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+import seedu.address.model.person.exceptions.DuplicateLabException;
+import seedu.address.model.person.exceptions.LabNotFoundException;
+
+public class LabList implements Iterable<Lab> {
+
+    private final ObservableList<Lab> internalList = FXCollections.observableArrayList();
+    private final ObservableList<Lab> internalUnmodifiableList =
+            FXCollections.unmodifiableObservableList(internalList);
+
+    /**
+     * Returns true if the list contains an equivalent lab as the given argument.
+     *
+     * @param toCheck The Lab to be checked
+     */
+    public boolean contains(Lab toCheck) {
+        requireNonNull(toCheck);
+        return internalList.stream().anyMatch(toCheck::isSameLab);
+    }
+
+    /**
+     * Adds a lab to the list.
+     * The person must not already exist in the list.
+     *
+     * @param toAdd The Lab that you want to add.
+     */
+    public void add(Lab toAdd) {
```

Comment:
Want to throw exception in signature?

Reply:
A bunch of other methods in this class too

Reply:
Sure I can add it in.

Reply:
i think this is mimicking the UniquePersonList/MasterLabList. They did not throw exceptions in the signature, so we should be fine

Reply:
Ah yes, @geetengtan is right, the reason why I didn't add the throw statements is that I followed the format of `UniquePersonList.java`. But I will still add them in for clarity. 😄